### DISCUSSION TITLE
File 'dnSpy-Unity-mono-v2020.x-V40' doesn't exist

### DISCUSSION THREAD

**Discussion by Soft-me** [dnSpyEx/dnSpy#124] [Q&A] [answered]
hmm , I try to use [dnSpy-Unity-mono](https://github.com/dnSpyEx/dnSpy-Unity-mono#supporting-a-new-unity-version-for-dummies) to build mono-2.0-bdwgc.dll for a Unity version 2020.2.2 .

So I follow all steps in [Supporting a new Unity version for Dummies](https://github.com/dnSpyEx/dnSpy-Unity-mono#supporting-a-new-unity-version-for-dummies).
In the last step:
Run umpatcher again to patch the code and commit it to the dnSpy-Unity-mono repo

`umpatcher 5.4.3 aa8a6e7afc2f4fe63921df4fe8a18cfd0a441d19 "C:\path\to\Unity-mono" "C:\path\to\dnSpy-Unity-mono"`

I get error: "File 'dnSpy-Unity-mono-v2020.x-V40' doesn't exist" during the execution of "Patching solution, projects and source code files"
I have to say I am a totally newbie for these things. Do I miss something or I completely do it in the wrong way?

**Comment by ElektroKill** [comment, marked as answer]
Hi, this is not entirely an error on your side. The dnSpy unity mono patcher doesn't yet officially support unity versions past 2019. The patcher requires a VS solution to be present for every major Unity version, in this case `2020`. This file does not exist in the repository yet as there haven't been any unity versions added with that major version number. An error you are making is specifying the incorrect version of Unity to the `umpatcher` program. The first argument specifies the version of Unity that you are currently trying to patch, this is not the reason for the error though. I will look into adding the necessary files for supporting newer major versions of Unity in the coming days.

**Reply by Soft-me** [reply]
OK !
Well, I am not familiar with this thing, otherwise I'd be happy to help and create a VS solution file.
Anyway, thanks for your replying! 
Hope the new necessary files will be added soon~
